Library/Homebrew/resource.rb

```diff
@@ -66,8 +66,19 @@ def escaped_name
     name.tr("/", "-")
   end
 
+  def self.safe_download_name(str)
+    str.gsub("@", "-AT-")
+  end
+
   def download_name
-    name.nil? ? owner.name : "#{owner.name}--#{escaped_name}"
+    # Must not have "@" in download_name because it's a special character to Subversion
+    Resource.safe_download_name(
+      if name
+        "#{owner.name}--#{escaped_name}"
+      else
```

Comment:
Feels weird having an if inside a function call. I’d assign a new variable outside it.

Reply:
Rewrote it to use a temporary variable. Moved & merged comment to `safe_download_name`.